sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala

```diff
@@ -303,7 +303,7 @@ package object dsl {
 
       def join(
         otherPlan: LogicalPlan,
-        joinType: JoinType = Inner,
+        joinType: JoinType = Inner(false),
```

Comment:
nit: not a big deal either way but explicitly naming the boolean parameters (`Inner(explicitCartesian = false)`) would be great for readability.